tools/proteore_filter_keywords_values/filter_kw_val.xml

```diff
@@ -0,0 +1,271 @@
+<tool id="MQoutputfilter" name="Filter by keywords and/or numerical value" version="2019.06.27">
+    <description></description>
+    <requirement type="package" version="3.7.3">python</requirement>
+    <exit_code range="1:" />
+    <command><![CDATA[
+        python '$__tool_directory__/filter_kw_val.py'
+        -i '$input1,$header'
+        -o '$kept_lines'
+        --discarded_lines '$discarded_lines'
+        --operation '$operation'
+        --operator '$operator'
+
+        ## Keywords
+        #for $key in $keyword
+            #if $key.k.kw != "None"
+                #if $key.k.kw == "text"
+                    --kw "$key.k.txt" "$key.ncol" "$key.match"
+                #else if $key.k.kw == "file"
+                    --kw_file "$key.k.file" "$key.k.header" "$key.k.ncol" "$key.ncol" "$key.match" 
+                #end if
+            #end if
+        #end for
+
+        ## value to filter
+        #for $val in $value
+            #if $val.value != "None"
+                --value $val.value $val.ncol $val.operator
+            #end if
+        #end for
+
+        ##range of values to keep
+        #for $vr in $values_range
+            #if vr 
+                --values_range $vr.bottom_value $vr.top_value $vr.ncol $vr.inclusive
+            #end if
+        #end for
+
+        #if $sort.sort_bool == "true"
+            --sort_col "$sort.sort_column,$sort.reversed_sort"
+        #end if
+
+    ]]></command>
+    <inputs>
+        <param type="data" name="input1" format="txt,tabular" label="Input file" />
+        <param name="header" type="boolean" checked="true" truevalue="true" falsevalue="false" label="Does file contain header?" />
+        <param name="operation" type="select" label="Operation" help="keep or discard word(s) or value(s) that match filters ?">
```

Comment:
The tool would become even more powerful if the  keep/discard could be selected for each filter, or? Maybe leaving this parameter as default.